src/main/java/br/com/openenade/api/universidade/UniversidadeService.java

```diff
@@ -13,68 +13,71 @@
 
-	public Universidade save(Universidade universidade) {
-		Optional<Municipio> optMunicipio = this.municipioService
-				.getOptionalByCodigo(universidade.getCampus().getCodigo());
-		Municipio newCampus;
-		if (optMunicipio.isPresent()) {
-			newCampus = optMunicipio.get();
-		} else {
-			newCampus = this.municipioService.save(universidade.getCampus());
-		}
-		universidade.setCampus(newCampus);
-		return this.repository.save(universidade);
+    public Universidade save(Universidade universidade) {
+        Optional<Municipio> optMunicipio =
+                this.municipioService.getOptionalByCodigo(universidade.getCampus().getCodigo());
+        Municipio newCampus;
+        if (optMunicipio.isPresent()) {
+            newCampus = optMunicipio.get();
+        } else {
+            newCampus = this.municipioService.save(universidade.getCampus());
+        }
+        universidade.setCampus(newCampus);
+        return this.repository.save(universidade);
 
-	}
+    }
 
-	public Collection<Universidade> getAll() {
-		return this.repository.findAll();
-	}
+    public Collection<Universidade> getAll() {
+        return this.repository.findAll();
+    }
 
-	// Issue #38
-	public Collection<Universidade> getAllByCodigoIES(Long codigoIES) {
+    // Issue #38
+    public Collection<Universidade> getAllByCodigoIES(Long codigoIES) {
 
-		return this.repository.findAllByCodigoIES(codigoIES);
+        return this.repository.findAllByCodigoIES(codigoIES);
 
-	}
+    }
 
-	public Universidade getUniversidadeByCodigoIES(Long codigoIES) {
-		Optional<Universidade> optUnivesidade = this.repository.findByCodigoIES(codigoIES);
-		if (optUnivesidade.isPresent()) {
-			return optUnivesidade.get();
-		} else {
-			throw new ResourceNotFound("" + codigoIES);
-		}
-	}
+    public Universidade getUniversidadeByCodigoIES(Long codigoIES) {
+        Optional<Universidade> optUnivesidade = this.repository.findByCodigoIES(codigoIES);
+        if (optUnivesidade.isPresent()) {
+            return optUnivesidade.get();
+        } else {
+            throw new ResourceNotFound("" + codigoIES);
+        }
+    }
 
-	public void deleteUniversidadeByCodigoIES(Long codigoIES) {
-		this.repository.deleteByCodigoIES(codigoIES);
-	}
+    public void deleteUniversidadeByCodigoIES(Long codigoIES) {
+        this.repository.deleteByCodigoIES(codigoIES);
+    }
 
-	public void deleteUniversidadesByMunicipioCodigo(Municipio campus) {
+    public void deleteUniversidadesByMunicipioCodigo(Municipio campus) {
 
-		this.repository.deleteAllByCampus(campus);
-	}
+        this.repository.deleteAllByCampus(campus);
+    }
 
-	public Optional<Universidade> getUniversidadeById(Long codigoIES, Long codigoMunicipio) {
+    public Optional<Universidade> getUniversidadeById(Long codigoIES, Long codigoMunicipio) {
 
-		Municipio municipio = this.municipioService.getMunicipioByCodigo(codigoMunicipio);
+        Optional<Municipio> municipio = this.municipioService.getOptionalByCodigo(codigoMunicipio);
```

Comment:
Why is necessary to check the existence of the Municipio? I think that the repository will throw some exception and this exception can be automatically translated to an Http Status Response.

Reply:
Praticamente a mesma resposta do outro review.

Reply:
Tem razão, resolvido ;)